File: c++/0121-best-time-to-buy-and-sell-stock/solution.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int result = 0;
        int left = 0, right = 1;

        while (right < prices.size()) {
            if (prices[left] < prices[right]) {
                result = max(result, prices[right] - prices[left]);
            } else {
                left = right;
            }

            ++right;
        }

        return result;
    }
};

File: java/0121-best-time-to-buy-and-sell-stock/Solution.java
public class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int left = 0, right = 1;

        while (right < prices.length) {
            if (prices[left] < prices[right]) {
                max = Math.max(max, prices[right] - prices[left]);
            } else {
                left = right;
            }

            ++right;
        }

        return max;
    }
}
